plugins/discourse-calendar/assets/javascripts/discourse/components/post-calendar.gjs

```diff
@@ -0,0 +1,377 @@
+            const eventDetail = { ...detail };
+            let from = moment.tz(detail.from, detail.timezone);
+            let to = moment.tz(detail.to, detail.timezone);
+            eventDetail.from = from.format("YYYY-MM-DD");
+            eventDetail.to = to.format("YYYY-MM-DD");
+            events.push(this._buildStandaloneEvent(eventDetail));
+          } else {
+            events.push(this._buildStandaloneEvent(detail));
+          }
+          break;
+      }
+    });
+
+    const formattedGroupedEvents = {};
+    groupedEvents.forEach((groupedEvent) => {
+      const minDate = this.isFullDay
+        ? moment(groupedEvent.from).format("YYYY-MM-DD")
+        : moment(groupedEvent.from).utc().startOf("day").toISOString();
+      const maxDate = this.isFullDay
+        ? moment(groupedEvent.to || groupedEvent.from).format("YYYY-MM-DD")
+        : moment(groupedEvent.to || groupedEvent.from)
+            .utc()
+            .endOf("day")
+            .toISOString();
+
+      const identifier = `${minDate}-${maxDate}`;
+      formattedGroupedEvents[identifier] = formattedGroupedEvents[
+        identifier
+      ] || {
+        from: minDate,
+        to: maxDate || minDate,
+        localEvents: {},
+      };
+
+      formattedGroupedEvents[identifier].localEvents[groupedEvent.name] =
+        formattedGroupedEvents[identifier].localEvents[groupedEvent.name] || {
+          users: [],
+        };
+
+      formattedGroupedEvents[identifier].localEvents[
+        groupedEvent.name
+      ].users.push.apply(
+        formattedGroupedEvents[identifier].localEvents[groupedEvent.name].users,
+        groupedEvent.users
+      );
+    });
+
+    Object.keys(formattedGroupedEvents).forEach((key) => {
+      const formattedGroupedEvent = formattedGroupedEvents[key];
+      this._buildGroupedEvents(formattedGroupedEvent).forEach((event) => {
+        events.push(event);
+      });
+    });
+
+    return events;
+  }
+
+  _buildEvent(detail) {
+    const event = this._buildEventObject(
+      detail.from
+        ? {
+            dateTime: moment(detail.from),
+            weeklyRecurring: detail.recurring === "1.weeks",
+          }
+        : null,
+      detail.to
+        ? {
+            dateTime: moment(detail.to),
+            weeklyRecurring: detail.recurring === "1.weeks",
+          }
+        : null
+    );
+
+    event.extendedProps = {};
+    if (detail.post_url) {
+      event.extendedProps.postUrl = getURL(detail.post_url);
+    } else if (detail.post_number) {
+      event.extendedProps.postNumber = detail.post_number;
+    } else {
+      event.classNames = ["holiday"];
+    }
+
+    if (detail.timezoneOffset) {
+      event.extendedProps.timezoneOffset = detail.timezoneOffset;
+    }
+
+    return event;
+  }
+
+  _buildEventObject(from, to) {
+    const hasTimeSpecified = (d) => {
+      if (!d) {
+        return false;
+      }
+      return d.hours() !== 0 || d.minutes() !== 0 || d.seconds() !== 0;
```

Comment:
```suggestion
      return d.hours() || d.minutes() || d.seconds();
```

If you don't want to do this fine, but I think you could do this since if any of them are not 0 this will be true...would be nice if moment had a `hasTime()` function or something